.docs/api/DATABASE.md

```diff
@@ -0,0 +1,410 @@
+important to build a new API with Symphony's needs in mind, which requires dynamic table and columns names.
+
+Hence, the API was designed with the following goals, in that order:
+
+1. Safety/security
+2. Ease of use
+3. Performance
+4. Almost 100% retro-compatible with LTS APIs.
+
+## Breaking changes in `3.0.0`
+
+Before `3.0.0`, all database operations resided in the `MySQL` class.
+This class as been renamed to `Database` in `3.0.0`.
+In order to make it easier to migrate to `3.0.0`, it still ships a `MySQL` class, but does not use it.
+In future versions, it will be removed, so make sure that you access everything via `Symphony::Database()` instance.
+
+There used to be two ways to execute SQL queries: `query()` and `fetch()`.
+`fetch()` was used when we wanted to get tabular results.
+In all other cases, `query()` was used.
+
+Light wrappers around those two methods did exist.
+Developers could either call `fetchCol()`, `fetchRow()` or `fetchVar()` to reduce the tabular data in a single call.
+This was less than optimal, because the signatures of those functions where hard to remember and error prone.
+Developers could also call `update()`, `insert()` and `delete()` to issue quick write statements.
+But again, this was not flexible.
+Those six functions still exists and worked as they used to, but they got refactored.
+This compatibility layer will be removed in future versions, so make sure to update your code with the new API.
+
+## API
+
+In order to execute SQL statements, the first thing to do is to create a `DatabaseStatement` object.
+The `Database` class provides factory methods that allows creation of all derived `DatabaseStatement` classes.
+Once the statement object is created, configuration and execution of the statement is left to the caller.
+This makes it possible to pass statements around before the statement gets executed.
+It also removes the needs to concat and parse string, since all this heavy lifting is done under the hood.
+
+The API uses chains of command that returns the current instance.
+This often eliminate the need to create statement variable and go strait to the result.
+
+In order to execute the statement, developers needs to call the `execute()` method, which returns
+a `DatabaseStatementResult` object.
+The result exposes a `success()` method and a direct access to the underlying `PDO` result.
+
+```php
+$stm = Symphony::Database()->statement('This is a raw SQL statement');
+$result = $stm->execute();
+if (!$result->success()) {
+    die('It did not worked');
+}
+```
+
+Note that if there was a SQL error, `PDO` is configured to throw.
+We wrap does errors in a `DatabaseStatementException` in order to provide more context.
+
+Here is the list of all factory methods, the type of object they create and their basic usage:
+
+### `select()` -> `DatabaseQuery`
+```php
+Symphony::Database()
+    ->select(['t.*'])
+    ->from('table')
```

Comment:
I think we should use metasyntactic variables in the documentation to make it easier to read and differentiate what is database jargon and users code.

wdyt @nitriques ? 😁

Reply:
I would not do it for all example, but for the most complex one, why not !

Reply:
Good for me ! 🍰